registry/handlers/app.go

```diff
@@ -596,24 +596,19 @@ func (app *App) configureSecret(configuration *configuration.Configuration) {
 func (app *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close() // ensure that request body is always closed.
 
-	// Instantiate an http context here so we can track the error codes
-	// returned by the request router.
-	ctx := defaultContextManager.context(app, w, r)
```

Comment:
It broke a middleware.

The context used to initialize the middleware (via [`repository.InitFunc`](https://godoc.org/github.com/docker/distribution/registry/middleware/registry#InitFunc)) is related to the context I passed in via `handlers.NewApp(...)`.

However, the context received by the [`Repository`](https://godoc.org/github.com/docker/distribution#Repository) `Manifests` function is the request context and doesn't have access to elements in that original context.

It's true that there is a reference to the app (https://github.com/docker/distribution/blob/1935c8d50b6b9d3fa8a5fcdb860cd3241a35e8df/registry/handlers/app.go#L766) but how do you access it once the context has been wrapped inside another context?